__tests__/auth.spec.js

```diff
@@ -70,6 +76,82 @@ describe('AUTH API ENDPOINTS', () => {
         });
     });
 
+    describe('POST PASSWORD RESET api/v1/auth/reset', () => {
+        it('should not send mail if user email is invalid', async () => {
+            const { status, body } = await server()
+                .post('/api/v1/auth/reset')
+                .send({ email: 'ba@ gmail' });
+
+            expect(status).toBe(422);
+            expect(body).toMatchSnapshot();
+        });
+
+        it('should not send mail if user is not found', async () => {
+            const { status, body } = await server()
+                .post('/api/v1/auth/reset')
+                .send({ email: 'bambam@gmail.com' });
+
+            expect(status).toBe(404);
+            expect(body).toMatchSnapshot();
+        });
+
+        it('should send email if user is valid', async () => {
+            const user = getUser();
+            await createUser(user);
+            emailReset = user.email;
+
+            /* sendResetMail(emailReset); */
+            const { status, body } = await server()
+                .post('/api/v1/auth/reset')
+                .send({ email: emailReset });
+
+            expect(status).toBe(200);
+            expect(Object.keys(body.data)).toMatchSnapshot();
+        });
+    });
+
+    describe('PATCH PASSWORD RESET api/v1/auth/reset', () => {
+        it('should not reset password if token is invalid', async () => {
+            const { status, body } = await server()
+                .patch('/api/v1/auth/reset')
+                .send({
+                    password: 'qwert12345',
+                    password_confirmation: 'qwert12345',
+                    token: 'resettoken'
+                });
+
+            expect(status).toBe(422);
+            expect(body).toMatchSnapshot();
+        });
+
+        it('should not reset password if password is invalid', async () => {
+            const { status, body } = await server()
+                .patch('/api/v1/auth/reset')
+                .send({
+                    password: 'qwert',
+                    password_confirmation: 'qwert12345',
+                    token: 'resettoken'
+                });
+
+            expect(status).toBe(422);
+            expect(body).toMatchSnapshot();
+        });
+
+        it('should reset password if all inputs are valid', async () => {
```

Comment:
Let's add another assertion here. We need to know the password was actually reset. Let's use `bcrypt` to  assert that the new password matches the encrypted version stored in the database. It would look something like this:

```js
expect(Bcrypt.compareSync(user.password, 'qwert12345')).toBe(true)
```

Reply:
done